api/Api/AssetManager/SynVisualElementAsset.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.IO;
+using System.Xml;
+using SynthesisAPI.AssetManager;
+using SynthesisAPI.UIManager;
+using SynthesisAPI.UIManager.VisualElements;
+using SynthesisAPI.VirtualFileSystem;
+
+namespace SynthesisAPI.AssetManager
+{
+    public class SynVisualElementAsset: Asset
+    {
+        private XmlDocument? _document;
+
+        public SynVisualElementAsset(string name, Permissions perms, string sourcePath)
+        {
+            Init(name, perms, sourcePath);
+        }
+
+        public SynVisualElement GetElement(string name)
+        {
+            if (_document == null)
+                Load(File.ReadAllBytes(FileSystem.BasePath + SourcePath));
```

Comment:
I don't think this is going to work because assets aren't really backed by files but by entries in a zip archive. So we'll need to find a different solution.